src/SocialPostManager.php

```diff
@@ -0,0 +1,336 @@
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   * @param string $token
+   *   Token to be used for autoposting.
+   * @param string $name
+   *   Name of user provided by social provider.
+   * @param string $additional_data
+   *   Additional data to be stored in record.
+   *
+   * @return bool
+   *   True if User record was created or False otherwise
+   */
+  public function addRecord($provider_user_id, $token, $name = '', $additional_data = '') {
+    // Get User ID of logged in user.
+    $user_id = $this->currentUser->id();
+    if ($this->checkIfUserExists($provider_user_id)) {
+      return FALSE;
+    }
+    // Encode token into json format.
+    $json_token = json_encode($token);
+    // Add user record.
+    $values = [
+      'user_id' => $user_id,
+      'plugin_id' => $this->pluginId,
+      'provider_user_id' => $provider_user_id,
+      'token' => $this->encryptToken($json_token),
+      'name' => $name,
+      'additional_data' => $additional_data,
+    ];
+
+    $user_info = $this->entityTypeManager->getStorage('social_post')->create($values);
+
+    // Save the entity.
+    $user_info->save();
+
+    if ($user_info) {
+      return TRUE;
+    }
+    else {
+      return FALSE;
+    }
+
+  }
+
+  /**
+   * Update token of a particular record.
+   *
+   * @param string $plugin_id
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   * @param string $token
+   *   Token provided by social_network.
+   *
+   * @return bool
+   *   True if updated else False otherwise.
+   */
+  public function updateToken($plugin_id, $provider_user_id, $token) {
+    $field_storage_configs = $this->entityTypeManager
+      ->getStorage('social_post')
+      ->loadByProperties(['plugin_id' => $plugin_id, 'provider_user_id' => $provider_user_id]);
+
+    $save_token = '';
+
+    foreach ($field_storage_configs as $field_storage) {
+      $field_storage->token = $token;
+      $field_storage->enforceIsNew(FALSE);
+      $save_token = $field_storage->save();
+    }
+
+    if ($save_token) {
+      return TRUE;
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * Used to get token for autoposting by implementers.
+   *
+   * @param string $plugin_id
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   *
+   * @return string
+   *   Token in array format.
+   */
+  public function getToken($provider_user_id) {
+
+    $storage = $this->entityTypeManager->getStorage('social_post');
+    // Perform query on social post entity.
+    $query = $this->entityQuery->get('social_post');
+
+    // Check user for social post implementer.
+    $user_data = current($this->entityTypeManager->getStorage('social_post')->loadByProperties(['plugin_id' => $this->pluginId, 'provider_user_id' => $provider_user_id]));
+
+
+    if (!$social_post_record) {
```

Comment:
This variable doesn't exist anymore. Please test your code before asking for review